pkg/cluster/cluster.go

```diff
@@ -427,13 +432,11 @@ func (c *Cluster) compareStatefulSetWith(statefulSet *appsv1.StatefulSet) *compa
 	}
 
 	if !reflect.DeepEqual(c.Statefulset.Spec.Template.Annotations, statefulSet.Spec.Template.Annotations) {
-		match = false
```

Comment:
why is `match`no longer necessary in all these conditions ?

Reply:
wherever `needsRollUpdate` or `needsReplace` are set to `true` we do not need to do the same for `match` because in the end we do:
```
	if needsRollUpdate || needsReplace {
		match = false
	}
```